### PR TITLE
Loguru

### PR DESCRIPTION
**Pull request by dorschw**
## Related Issues
fixes: https://jira-dc.paloaltonetworks.com/browse/CIAC-11561

## Description
Use [Loguru](https://github.com/Delgan/loguru) instead of the builtin logging

### CODE REVIEW

File: demisto_sdk/commands/common/loguru_logger.py [hunk 1]
```diff
@@ -0,0 +1,177 @@
+import os
+import platform
+import sys
+from pathlib import Path
+from typing import Any, Iterable, Optional, Union
+
+import loguru  # noqa: TID251 # This is the only place where we allow it
```
Comment (tkatzir) [inline demisto_sdk/commands/common/loguru_logger.py:7]: Do we have a validation that TID251 isn’t used in any other file?

File: demisto_sdk/commands/common/logger.py [hunk 2]
```diff
@@ -1,487 +1,19 @@
...
-NO_COLOR_ESCAPE_CHAR = "\033[0m"
-
 DEMISTO_LOG_ALLOWED_ESCAPES = [  # The order of the list is by priority.
     ("green", 32),
     ("red", 91),
     ("yellow", 93),
     ("cyan", 36),
     ("blue", 34),
     ("orange", 33),
-    ("pink", 95),
```
Comment (tkatzir) [inline demisto_sdk/commands/common/logger.py:142]: I would leave these, for reference.

File: demisto_sdk/commands/common/loguru_logger.py [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+
+    raise ValueError(f"cannot convert {input_} to bool")
+
+
+global logger
+logger = loguru.logger  # all SDK modules should import from this file, not from loguru
+
+
+def setup_neo4j_logger():
+    # TODO use
```
Comment (tkatzir) [inline demisto_sdk/commands/common/loguru_logger.py:44]: ???

File: demisto_sdk/commands/common/loguru_logger.py [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+            return f"{int(env_var)} B"
+
+        except (TypeError, ValueError):
+            logger.warning(
+                f"Invalid value for DEMISTO_SDK_LOG_FILE_SIZE environment variable: {env_var}. Using default value of '1MB'."
+            )
+    return "1 MB"
+
+
+def calculate_rentation() -> int:
```
Comment (tkatzir) [inline demisto_sdk/commands/common/loguru_logger.py:63]: `rentation`?!?
```suggestion
def calculate_log_files_count() -> int:
```

File: pyproject.toml [hunk 5]
```diff
@@ -179,7 +181,7 @@ select = [
 [tool.ruff.lint.flake8-tidy-imports.banned-api]
 "distutils.version".msg = "Use packaging.version.Version instead"
 "git.Repo".msg = "Use GitUtil instead"
-# "logging".msg = "The logging framework is banned. Please use demisto_sdk.commands.common.logger."
+# "logging".msg = "The logging framework is banned. Please use demisto_sdk.commands.common.logger." TODO
```
Comment (tkatzir) [inline pyproject.toml:184]: ???

**Review by tkatzir — COMMENTED** [review]
Very good!
Some comments.

File: demisto_sdk/commands/common/loguru_logger.py (hunk 1 above)
Comment (dorschw) [reply demisto_sdk/commands/common/loguru_logger.py:7]: Don't think it's possible

**Comment by dorschw** [comment]
replaced by #4514

**Pull request closed without merging**